log4j-core/src/main/java/org/apache/logging/log4j/core/selector/ContextSelector.java

```diff
@@ -108,7 +108,7 @@ default LoggerContext getContext(String fqcn, ClassLoader loader, Map.Entry<Stri
     default LoggerContext getContext(String fqcn, ClassLoader loader, Map.Entry<String, Object> entry,
             boolean currentContext, URI configLocation) {
         final LoggerContext lc = getContext(fqcn, loader, currentContext, configLocation);
-        if (lc != null) {
+        if (lc != null && entry != null) {
```

Comment:
`lc` should be not `null` here, we just need to check if `entry` is not `null`.

Reply:
Since `getContext()` is to be implemented by the subclass, I thought we cannot have such an assumption. Nevertheless, if you think otherwise, be my guest to update it.